spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -3179,17 +3260,35 @@ def leaveEvent(self, event):
             QApplication.restoreOverrideCursor()
             self.__cursor_changed = False
             self.clear_extra_selections('ctrl_click')
+            self._last_hover_uri = None
         TextEditBaseWidget.leaveEvent(self, event)
 
     def mousePressEvent(self, event):
-        """Reimplement Qt method"""
+        """Override Qt method."""
         ctrl = event.modifiers() & Qt.ControlModifier
         alt = event.modifiers() & Qt.AltModifier
         pos = event.pos()
         if event.button() == Qt.LeftButton and ctrl:
             TextEditBaseWidget.mousePressEvent(self, event)
             cursor = self.cursorForPosition(pos)
-            self.go_to_definition_from_cursor(cursor)
+
+            if self._last_hover_uri:
+                uri = self._last_hover_uri
+                if uri.startswith('file://'):

```

Comment:
The logic to open files is already handled in the `open_file` method of `MainWindow`. I think we should use that instead.

Reply:
I do not understand

Reply:
Yeah, this is too convoluted. I left a comment below.